src/Raven.Server/Documents/Sharding/Handlers/Processors/Subscriptions/ShardedSubscriptionsHandlerProcessorHelper.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Changes;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Client.Exceptions.Sharding;
+using Raven.Server.Documents.Replication;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.Documents.TcpHandlers;
+using Raven.Server.ServerWide.Commands.Sharding;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.TrafficWatch;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.Subscriptions
+{
+    internal static class ShardedSubscriptionsHandlerProcessorHelper
+    {
+        private struct ChangeVectorValidationResult
+        {
+            public Dictionary<string, string> ChangeVectorsCollection;
+            public string InitialChangeVector;
+        }
+
+        public static async Task CreateSubscriptionInternalAsync(ShardedSubscriptionsHandler requestHandler, BlittableJsonReaderObject bjro, long? id, bool? disabled, SubscriptionCreationOptions options, JsonOperationContext context)
```

Comment:
can we rename this to `CreateSubscriptionAsync`

Reply:
In general I would love to unify this with the regular database method.
Then the proper place for that would be in the `ShardedSubscriptions` class.

Reply:
done